### PR TITLE
[JENKINS-54599] - Upgrade the Maven Jenkins Dev plugin from 9.4.5.v20170502 to 9.4.12.v20180830

### PR DESCRIPTION
**Pull request by oleg-nenashev**
See [JENKINS-54599](https://issues.jenkins-ci.org/browse/JENKINS-54599). This is a downstream of https://github.com/jenkinsci/maven-hudson-dev-plugin/pull/9 

### Proposed changelog entries

* RFE: Internal: Upgrade the Maven Jenkins Dev plugin from 9.4.5.v20170502 to 9.4.12.v20180830 to align it with the Jetty version 

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jenkinsci/java11-support

### CODE REVIEW

**Review by alecharp — APPROVED** [review]
LGTM

**Review by batmat — APPROVED** [review]
LGTM, provided I expect you'll bump to an actual release before merging right? For now I suppose your intent was to check the core would still build with latest version of the plugin using a timestamped version before actually using release version for the modified Maven Jenkins Dev plugin.

**Review by batmat — APPROVED** [review]

**Pull request merged**
